### ISSUE TITLE
Overlay not working correct - choppy sound

### ISSUE THREAD

**Issue by jerearaujo03** [jiaaro/pydub#494]
I have two mono channels (they are a stream buffer, not file) I want both to play simultaneous in stereo, but it's not working.
If I play each one alone, they sound well, but when I do `channel1.overlay(channel2)` they sound choppy.

My code is
```
channel1 = AudioSegment(bytes(channels[1]), sample_width=2, frame_rate=48000, channels=1)
channel1 = channel1.set_channels(2)

channel2 = AudioSegment(bytes(channels[2]), sample_width=2, frame_rate=48000, channels=1)
channel2 = channel2.set_channels(2)
```
and I'm mixing doing
```
mix1 = channel1.overlay(channel2)
```

then if I do `print(len(mix1).get_array_of_samples())` I get **480**, but if I do `print(len(channel1.get_array_of_samples()))` I get **512** (same for channel2)

Why is it happening?

**Comment by GreyAlien502** [comment]
The problem is that ```overlay``` rounds lengths to the nearest millisecond.

**Comment by prohilla** [comment]
@jerearaujo03 did you by any chance figure out a fix or workaround? I am facing the same issue.
@jiaaro Will this be addressed anytime soon? Is it something quick to fix or is there any workaround we can follow?

**Comment by jerearaujo03** [comment]
@prohilla no, I was not able to make it work. I ended up doing the mix manually, averaging the sample values

**Comment by jiaaro** [comment]
pydub isn't really designed to work on hundreths of a second, it assumes you are working with "human" lengths of audio and that if you have a longer piece of sound, that you'll operate on it in full, rather than slicing it into hundreds or thousands of pieces. For cases when you _do_ want to work on hundreds of samples rather than tens of thousands, that is exactly what the `get_array_of_samples()`API is for. Perhaps though there is room to tighten up the overlay api so that AudioSegment objects do no round to the nearest millisecond. Pull requests welcome, as they say 😁

**Comment by prohilla** [comment]
> @prohilla no, I was not able to make it work. I ended up doing the mix manually, averaging the sample values

Thanks a lot for your response @jerearaujo03. Do you mind sharing how to mix manually? Will be really helpful. I am new to pydub.

**Comment by prohilla** [comment]
> pydub's AudioSegment isn't really designed to work on hundreths of a second – it assumes you're working with "human" lengths of audio and that if you have a longer piece of sound, that you'll operate on it in full, rather than slicing it into hundreds or thousands of pieces.
> 
> For cases when you _do_ want to work on hundreds of samples rather than tens of thousands, that is exactly what the `get_array_of_samples()`API is for. Perhaps though there is room to tighten up the overlay api so that AudioSegment objects do no round to the nearest millisecond. Pull requests welcome, as they say 😁

@jiaaro Thanks a lot for your reply. I am using pydub in a poc project currently. Once the POC turns into a full project, I will make my team look into it.
